server/aesdsocket.c

```diff
@@ -0,0 +1,340 @@
+            outfile_buff_index += bytes_to_write;
+
+            if (newline_found) {
+                //Write out packet
+                bytes_written_to_file = write(outfile_fd, &outfile_buff[0], outfile_buff_index);
+                if (bytes_written_to_file == -1) {
+                    perror("write");
+                    syslog(LOG_ERR, "write");
+                    close(outfile_fd);
+                    close(client_fd);
+                    close(sock_fd);
+                    free(outfile_buff);
+                    exit(EXIT_FAILURE);
+                }
+
+                outfile_size += bytes_written_to_file;
+                //Send full file contents
+                lseek(outfile_fd, 0, SEEK_SET);
+                
+                while (bytes_written_to_client < outfile_size) {
+                    if (bytes_written_to_client + FILE_READOUT_CHUNK > outfile_size) {
+                        readout_write_chunk = outfile_size - bytes_read_from_file;
+                    }
+                    else {
+                        readout_write_chunk = FILE_READOUT_CHUNK;
+                    }
+
+                    readout_read_chunk = read(outfile_fd, &file_readout_buff[0], readout_write_chunk);
+                    if (bytes_recvd == -1) {
+                        perror("read");
+                        syslog(LOG_ERR, "read");
+                        free(outfile_buff);
+                        close(outfile_fd);
+                        close(client_fd);
+                        close(sock_fd);
+                        exit(EXIT_FAILURE);
+                    }
+                    bytes_read_from_file += readout_read_chunk;
+
+                    bytes_to_write = send(client_fd, &file_readout_buff[0], readout_read_chunk, 0);
+                    if (bytes_to_write == -1) {
+                        perror("send");
+                        syslog(LOG_ERR, "send");
+                        free(outfile_buff);
+                        close(outfile_fd);
+                        close(client_fd);
+                        close(sock_fd);
+                        exit(EXIT_FAILURE); 
+                    }
+
+                    bytes_written_to_client += readout_write_chunk;
+                }
+                bytes_written_to_client = 0;
+                bytes_read_from_file = 0;
+                readout_read_chunk = 0;
+                readout_write_chunk = 0;
+
+                lseek(outfile_fd, 0, SEEK_END);
+            }
+            
+            recv_buff_index = 0;
+            newline_found = 0;
+        }
+
+        if (close(client_fd) == -1) {
+            perror("close");
+            syslog(LOG_ERR, "close");
+            free(outfile_buff);
+            close(sock_fd);
+            close(outfile_fd);
+            exit(EXIT_FAILURE); 
+        }
+
+        syslog(LOG_INFO, "Closing connection from %s", ip_addr_str);
+
+        if (sigprocmask(SIG_UNBLOCK, &sig_mask, NULL)) {
+            syslog(LOG_ERR, "sigprocmask");
```

Comment:
@vido2373 missing cleanup here.  Consider a design where you set a return value but allow the main cleanup path to run in success or failure.  This will avoid duplicating logic in all of your failure exit handlers.